src/vsr/replica.zig

```diff
@@ -5278,6 +5280,14 @@ pub fn ReplicaType(
 
             if (self.status == .view_change and self.primary_index(self.view) == self.replica) {
                 if (self.journal.dirty.count == 0 and self.commit_min == self.commit_max) {
+                    if (self.commit_stage != .idle) {
+                        // If we still have a commit running, we started it the last time we were
+                        // primary, and its still running. Wait for it to finish before repairing
+                        // the pipeline so that it doesn't wind up in the new pipeline.
+                        assert(self.commit_prepare.?.header.op == self.commit_min + 1);
```

Comment:
We don't bother to kick `.repair` eagerly after we transition to `.idle` and just wait for repair timeout here, because this is an edge case, right?

Reply:
> We don't bother to kick .repair eagerly after we transition to .idle and just wait for repair timeout here,

I'm not sure what you mean -- we are already in `repair()` right now, and I don't think we need to repair anyway (we already have the prepare and are committing it).

Reply:
I mean that when we finish committing, we could trigger repair immediately

Reply:
Ah, gotcha. Yeah, since this is such a rare case I think it's fine to wait for the `repair_timeout` to move things forward.